File: iterator1.py
c = ["Nikolaos", "Maria", "Petros", "Sofia"]

# διάσχιση λίστας με iterator
itr = iter(c)
while True:
    try:
        print(next(itr))
    except StopIteration:
        break

print("#" * 20)

# απόκρυψη του iterator
for x in c:
    print(x)

# Nikolaos
# Maria
# Petros
# Sofia
# ####################
# Nikolaos
# Maria
# Petros
# Sofia

File: iterator1.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    vector<string> v{"Nikolaos", "Maria", "Petros", "Sofia"};
    
    // διάσχιση vector με iterator
    vector<string>::iterator itr = v.begin();
    while (itr != v.end())
    {
        cout << *itr << endl;
        itr++;
    }

    cout << "################################" << endl;

    // απόκρυψη του iterator
    for (const auto &x : v)
    {
        cout << x << endl;
    }
}

// Nikolaos
// Maria
// Petros
// Sofia
// ################################
// Nikolaos
// Maria
// Petros
// Sofia